### DISCUSSION TITLE
Viscosity enrollment

### DISCUSSION THREAD

**Discussion by gaturpin** [PrincetonUniversity/athena#411] [Q&A] [answered]
Hey,

I'm trying to implement an alpha viscosity disc by enrolling my own viscosity function:

```
void Mesh::InitUserMeshData(ParameterInput *pin) {
  if (pin->GetReal("problem","nu_aniso")>0){
    alpha_disc = pin->GetReal("problem", "alpha_disc");
    EnrollViscosityCoefficient(Viscosity); // alpha viscosity
  }
}

void Viscosity(HydroDiffusion *phdif, MeshBlock *pmb, 
              const AthenaArray<Real> &prim,const AthenaArray<Real> &bcc, 
              int is, int ie, int js, int je,int ks, int ke){

  Real x, y, z, r, sound_speed;
  if (phdif->nu_aniso > 0.0){
    for (int k=ks; k<=ke; ++k) {
      for (int j=js; j<=je; ++j) {
#pragma omp simd
        for (int i=is; i<=ie; ++i){
          x = pmb->pcoord->x1v(i);
          y = pmb->pcoord->x2v(j);
          z = pmb->pcoord->x3v(k);
          r = std::sqrt(x*x + y*y);
          sound_speed = std::sqrt(SoundSpeedSQR(x,y,z));
          Real nu_viscosity = alpha_disc*sound_speed*h_over_r*r;
          phdif->nu(HydroDiffusion::DiffProcess::aniso,k,j,i) = nu_viscosity;
        }
      }
    }
  }
}
```

The function seems to be doing what I expect, however there isn't any change in the outputs. Any suggestions on where I might be going wrong or what I might be missing?

I understand that nu_aniso will be read by the code and nothing will be done with it, but would this not be the right thing to do in order to have a nu viscosity that varies over the domain?

Cheers!

**Comment by pdmullen** [comment, marked as answer]
Hi @gaturpin 😄 

Sorry if I am missing your question... but I think the answer here is that anisotropic viscosity is not yet implemented in Athena++.  If you change `phdif->nu(HydroDiffusion::DiffProcess::aniso,k,j,i) = nu_viscosity;` to `phdif->nu(HydroDiffusion::DiffProcess::iso,k,j,i) = nu_viscosity;`, then I believe you will get the behavior you are after.  You can read more about the available diffusive physics in Athena++ here: https://github.com/PrincetonUniversity/athena/wiki/Diffusion-Processes

Sincerely,
Patrick

**Comment by c-white** [comment]
To add to Patrick's comment, there might be some confusion about "anisotropic." Athena uses this to mean diffusion is different in different directions, not that diffusion is different in different locations.

**Comment by gaturpin** [comment]
Great, thank you! I did start doing this but it felt weird to me because, as Chris said, I was confused by iso and aniso in this context.

Thanks both (@pdmullen @c-white) for clearing this up!
